packages/transport-bluetooth/src/client/trezor-bluetooth.ts

```diff
@@ -0,0 +1,75 @@
+import { WebsocketClient } from '@trezor/websocket-client';
+
+import {
+    BluetoothDevice,
+    BluetoothInfo,
+    Logger,
+    NotificationEvent,
+    TrezorBluetoothSettings,
+} from './types';
+
+// Client for trezor-bluetooth rust websocket server
+export class TrezorBluetooth extends WebsocketClient<NotificationEvent> {
+    readonly settings: TrezorBluetoothSettings;
+    readonly logger: Logger;
+
+    constructor(settings: TrezorBluetoothSettings) {
+        super({
+            url: settings.url,
+            timeout: settings.timeout || 60 * 1000,
+        });
+        this.settings = Object.freeze(settings);
+        this.logger = settings.logger || {
+            info: () => {},
+            debug: () => {},
+            log: () => {},
+            warn: () => {},
+            error: () => {},
+        };
+    }
+
+    createWebsocket() {
+        return this.initWebsocket({
+            url: this.settings.url,
+            headers: {
+                Origin: 'https://node.trezor.io',
+                'User-Agent': 'Trezor Suite',
+            },
+            timeout: this.settings.timeout || 60 * 1000,
+        });
+    }
+
+    ping() {
+        return Promise.resolve(this.sendRawMessage('PING'));
+    }
+
+    send(method: 'get_info', adapter?: boolean): Promise<BluetoothInfo>;
+    send(method: 'enumerate'): Promise<BluetoothDevice[]>;
+    send(method: 'start_scan'): Promise<BluetoothDevice[]>;
```

Comment:
I wonder what is the difference between enumerate and start_scan. they have the same signature

Reply:
`enumerate` will just return known BT adapter state, BT adapter doesnt send scan requests, adapter is not subscribed to scan events. basically without scanning you will receive only paired devices even if they not around (works on linux and windows, macos returns nothing)

`start_scan` will start long term scanning process in the background which eventually should emit some scan events (device discovered, connected, disconnected etc...) until stopped by `stop_scan`
method returns `enumerate` but after(!) scanning, so the device list could be extended with scan results

Reply:
ok, thanks for explanation